kotlin/android/app/src/main/java/dev/firezone/android/tunnel/util/DnsServersDetector.kt

```diff
@@ -0,0 +1,239 @@
+package dev.firezone.android.tunnel.util
+
+import android.annotation.TargetApi
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.LinkProperties
+import android.os.Build
+import android.util.Log
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import java.io.LineNumberReader
+import java.net.InetAddress
+
+/**
+ * DNS servers detector
+ *
+ * IMPORTANT: don't cache the result.
+ *
+ * Or if you want to cache the result make sure you invalidate the cache
+ * on any network change.
+ *
+ * It is always better to use a new instance of the detector when you need
+ * current DNS servers otherwise you may get into troubles because of invalid/changed
+ * DNS servers.
+ *
+ * This class combines various methods and solutions from:
+ * Dnsjava http://www.xbill.org/dnsjava/
+ * Minidns https://github.com/MiniDNS/minidns
+ * https://stackoverflow.com/a/48973823/1275497
+ *
+ * Unfortunately both libraries are not aware of Orero changes so new method was added to fix this.
+ *
+ * Created by Madalin Grigore-Enescu on 2/24/18.
```

Comment:
@pratikvelani Since this was copied from another project / author we need to properly attribute and include [its MIT license](https://github.com/Willena/OpenDNSUpdater/blob/master/LICENSE) in our top-level `NOTICE.txt`.

Reply:
Fixed